ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala

```diff
@@ -615,4 +617,33 @@ private[appendonlydao] object TransactionsReader {
       }
       ._3
   }
+
+  /** Groups together items of type [[A]] that share an attribute [[K]] over a
+    * contiguous stretch of the input [[Source]]. Well suited to perform group-by
+    * operations of streams where [[K]] attributes are either sorted or at least
+    * show up in blocks.
+    *
+    * Implementation detail: this method _must_ use concatSubstreams instead of
+    * mergeSubstreams to prevent the substreams to be processed in parallel,
+    * potentially causing the outputs to be delivered in a different order.
+    *
+    * Docs: https://doc.akka.io/docs/akka/2.6.10/stream/stream-substream.html#groupby
+    */
+  def groupContiguous[A, K, Mat](
+      source: Source[A, Mat]
+  )(by: A => K): Source[Vector[A], Mat] =
+    source
+      .statefulMapConcat(() => {
+        var previousSegmentKey: K = null.asInstanceOf[K]
```

Comment:
Do you need to type-cast it? Isn't Null subtype of everything?

Reply:
`AnyRef` indeed. `AnyVals` can't be null

Reply:
@mziolekda restricting breaks code. It works without restriction as well. Let's just leave this as is (please not this is not new code, just moved around)

Reply:
I didn't mix them, but afaik type erasure implies `Object` (aka `AnyRef`). That's what I meant by anything. Probably I should've been more precise

Reply:
I understand that this code is moved. git blame won't know that though :-P. You might want to wrap previousSegmentKey in an option instead. Up to you.

Reply:
Hmm, wait, it should only box if you let the compiler decide (e.g. `val k: T = _`). Did it actually compile?

Reply:
Yes, that was my original point sorry for being not entirely clear. 
I kind a like that no extra objects are created.
But just to prevent spending further time on this I will switch it to an Option :+1: